content/en/blog/_posts/Hardware-Accelerated-SSLTLS-Termination-in-Ingress-Controllers-using-Kubernetes-Device-Plugins-and-RuntimeClass.md

```diff
@@ -0,0 +1,168 @@
+through to the container as hardware isolated devices. The only requirement is that the
+Kata Containers kernel has driver for the exposed device enabled.
+
+That’s all it really takes to enable hardware accelerated crypto for container workloads. To summarize:
+
+  1. Cluster needs a device plugin running on the node that provides the hardware,
+  2. Device plugin exposes the hardware to user space using  the VFIO driver,
+  3. Pod requests the device resources and Kata Containers as the RuntimeClass in the PodSpec,
+  4. The container has the hardware adaptation library and the OpenSSL engine module 
+
+Figure 2 shows the overall setup using the Container A illustrated earlier.
+
+{{<figure width="600"  src="/images/blog/2019-XX-YY-foo/k8s-blog-fig2.png" caption="Figure 2. Deployment overview">}}  
+
+## Reference Setup
+
+Finally, we describe the necessary building blocks and steps to build a functional
+setup described in Figure 2 that enables hardware accelerated SSL termination in
+an Ingress Controller using an Intel&reg; QuickAssist Technology (QAT) PCIe device.
+It should be noted that the use cases are not limited to Ingress controllers, but
+any OpenSSL based workload can be accelerated.
+
+### Cluster configuration:
+  * Kubernetes 1.14 (`RuntimeClass` and `DevicePlugin` feature gates enabled (both are `true` in 1.14)
+  * RuntimeClass ready runtime and Kata Containers configured 
+
+### Host configuration:
+  * Intel&reg; QAT driver release with the kernel drivers installed for both host kernel and Kata Containers kernel (or on a rootfs as loadable modules)
+  * [QAT device plugin](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/cmd/qat_plugin) DaemonSet deployed
+
+### Ingress controller configuration and deployment:
+  * [HAproxy-ingress](https://github.com/jcmoraisjr/haproxy-ingress)* ingress controller in a modified container that has
+     * the QAT HW HAL user space library (part of Intel&reg; QAT SW release) and
+     * the [OpenSSL QAT Engine](https://github.com/intel/QAT_Engine/) built in
+  * Haproxy-ingress ConfigMap to enable QAT engine usage
+     * `ssl-engine=”qat”`
+     * `ssl-mode-async=true`
+  * Haproxy-ingress deployment `.yaml` to
+     * Request `qat.intel.com: n` resources
+     * Request `runtimeClassName: kata-containers` (name value depends on cluster config)
+  * (QAT device config file for each requested device resource with OpenSSL engine configured available in the container)
+
+Once the building blocks are available, the hardware accelerated SSL/TLS can be tested by following the [TLS termination
+example](https://github.com/jcmoraisjr/haproxy-ingress/tree/master/examples/tls-termination) steps. In order to verify the hardware is used, you can check `/sys/kernel/debug/*/fw_counters` files on host as they
+get updated by the Intel&reg; QAT firmware.
+
+*Haproxy-ingress and HAproxy are used because HAproxy can be directly configured to use the OpenSSL engine using
```

Comment:
This `*` looks a little like the start of Markdown emphasis.

Something like:
```markdown
*<span/>Haproxy-ingress and HAproxy are used…
```

might help if emphsis isn't wanted.

Reply:
@sftim I saw this too after I submitted the PR. The preview shows this just fine.

Reply:
hmm ```<span/``` does not seem to have any effect.